napari/components/dims.py

```diff
@@ -154,24 +165,37 @@ def ndim(self, ndim):
     def indices(self):
         """Tuple of slice objects for slicing arrays on each dimension."""
         slice_list = []
-        z = zip(self.mode, self.display, self.point, self.interval)
-        for (mode, display, point, interval) in z:
-            if mode == DimsMode.POINT:
-                if display:
-                    slice_list.append(slice(None, None, None))
-                else:
-                    slice_list.append(int(round(point)))
-            elif mode == DimsMode.INTERVAL:
-                if display:
-                    slice_list.append(slice(None, None, None))
-
-                else:
-                    slice_list.append(
-                        slice(int(round(interval[0])), int(round(interval[1])))
-                    )
-
+        for axis in range(self.ndim):
+            if axis in self.displayed:
+                slice_list.append(slice(None))
+            else:
+                slice_list.append(int(round(self.point[axis])))
         return tuple(slice_list)
 
+    @property
+    def ndisplay(self):
+        """Int: Number of displayed dimensions."""
+        return self._ndisplay
+
+    @ndisplay.setter
+    def ndisplay(self, ndisplay):
+        if ndisplay not in (2, 3):
+            raise ValueError(
+                f"Invalid number of dimensions to be displayed {ndisplay}"
+            )
+        self._ndisplay = ndisplay
+        self.events.display()
+
+    @property
+    def displayed(self):
+        """tuple: list of displayed dimensions."""
+        return tuple(self.order[-self.ndisplay :])
+
+    @property
+    def not_displayed(self):
+        """tuple: list of dimensions not displayed."""
+        return tuple(self.order[: -self.ndisplay])
```

Comment:
also since this does not really need to be calculated on the fly you should set it whenever you update `self.ndisplay` and `self.order`